pepagent/pepagent.py

```diff
@@ -242,40 +287,64 @@ def get_projects(
             results = self.run_sql_fetchall(sql_q, *flattened_registries)
 
         # Case 3. Get projects by namespace
+        elif namespace:
+            if tag:
+                sql_q = (
+                    f"select {NAME_COL}, {PROJ_COL} "
+                    f"from {DB_TABLE_NAME} "
+                    f"where namespace = %s and tag = %s"
+                )
+                results = self.run_sql_fetchall(sql_q, namespace, tag)
+            else:
+                sql_q = f"select {NAME_COL}, {PROJ_COL} from {DB_TABLE_NAME} where namespace = %s"
+                results = self.run_sql_fetchall(sql_q, namespace)
+
+        # Case 4. Get projects by namespace
+        elif tag:
+            sql_q = f"select {NAME_COL}, {PROJ_COL} from {DB_TABLE_NAME} where tag = %s"
+            results = self.run_sql_fetchall(sql_q, tag)
+            print(results)
+
         else:
-            sql_q = f"select {NAME_COL}, {PROJ_COL} from {DB_TABLE_NAME} where namespace = %s"
-            results = self.run_sql_fetchall(sql_q, namespace)
+            _LOGGER.warning(f"Incorrect input!")
+            results = []
 
         # extract out the project config dictionary from the query
         return [peppy.Project(project_dict=p[1]) for p in results]
 
     def get_namespace(self, namespace: str) -> dict:
```

Comment:
I don't understand what it means to get a namespace, especially when you're giving a namespace as input. 

Probably this function needs a different name, and also explanation in docstring.

Reply:
I think this is akin to fetching metadata about repositories in dockerhub or github. This should return information like:
- What projects exist in this namespace?
- Links to project registries,
- How many samples in each project?
- Does the namespace have a description?

For pephub, I'd use this function to populate the data on a page like this: http://pephub.databio.org/pep/demo/view

Reply:
I think this is just a function name issue. Is `get_namespace` really what the function is doing? Maybe `get_namespace_metadata` ? Not sure if I'm overthinking this...

Reply:
I think this is just a function name issue. Is `get_namespace` really what the function is doing? Maybe `get_namespace_metadata` ? Not sure if I'm overthinking this...

Reply:
maybe: 'get_namespace_info' or 'get_namespace_project_info'. Because we are retrieving information about projects in the namespace. What do you think @nleroy917 ?

Reply:
I'm partial to `get_namespace_info`

Reply:
`get_namespace_info` is fine with me.